src/praisonai-bot/praisonai_bot/scheduler/executor.py

```
@@ -461,6 +481,131 @@ async def _execute_one(self, job: "ScheduleJob") -> JobResult:
+        except subprocess.TimeoutExpired:
+            if proc is not None:
+                try:
+                    if _POSIX:
+                        os.killpg(os.getpgid(proc.pid), 9)
+                    else:  # pragma: no cover - non-POSIX
+                        proc.kill()
+                except (ProcessLookupError, PermissionError, OSError):  # pragma: no cover
+                    proc.kill()
+                try:
+                    partial, _ = proc.communicate(timeout=5)
+                except Exception:  # pragma: no cover - best-effort reap
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)